server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -361,7 +363,8 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
 
     private final AtomicLong idGenerator = new AtomicLong();
 
-    private final Map<Long, ReaderContext> activeReaders = ConcurrentCollections.newConcurrentMapWithAggressiveConcurrency();
+    private final Map<ShardSearchContextId, ReaderContext> activeReaders = ConcurrentCollections
```

Comment:
Can we avoid this change? Hashing and comparing a ShardSearchContextId is more expensive than a Long.